redash/utils/json.py

```diff
@@ -0,0 +1,67 @@
+import json
+import binascii
+import decimal
+import uuid
+from sqlalchemy.orm.query import Query
+from datetime import timedelta, datetime
+from redash.query_runner import query_runners
+
+class JSONEncoder(json.JSONEncoder):
+    """Adapter for `json.dumps`."""
+
+    def __init__(self, **kwargs):
+        self.encoders = [
```

Comment:
Just curious why is this an array? Is it possible to have result from multiple query runners at this point?

Reply:
this is not a result, these are custom json encoder functions for each query runner

Reply:
How is there control then on what custom_json_encoder is then actually used? For example, pg.py and mognodb.py both have custom_json_encoder defined. Then self.encoders will be [encoder_from_pg, encoder_from_mongodb].

In that case, say we had a query result from mongo which needed to be encoded, wouldn't the following code try to use both encoders, one by one and say if both encoders had different logic for encoding Timestamp, whichever encoder is first in list get to do the encoding?

```
 def default(self, o):
        for encoder in self.encoders:
            result = encoder(self, o)
            if result:
                return result
```

Reply:
doesn't matter which encoder is first, cause custom encoders are to process custom data types, provided by runner's lib, like ObjectId for mongodb

Reply:
I've pushed a fix for #6747

Reply:
i don't get how your change isn't recursive? if encoder(self, o) returned an encoded string, then none of the elif condition in default(...) catch it, which could result in super().default(o) called again?

Reply:
thanks for clarification : )